scripts/get_previous_timestamp.py

```diff
@@ -35,24 +41,38 @@ def get_workflow_runs(token: str, repo: str):
               "per_page": 100}
     url = f"https://api.github.com/repos/{repo}/actions/runs"
     r = requests.get(url, params)
-    runs = json.loads(r.text)
+    if r.status_code >= 500:
+        with open("patchwork_down.txt", "w") as f:
+            f.write(f"status code: {r.status_code}")
+        return None
+    runs = [run for run in json.loads(r.text) if run["name"] == workflow]
+    print(f"After filter have {len(runs)} to consider")
     return runs
 
 def get_run_timestamp(runs, run_id: str):
-    runs = [run for run in runs if run["name"] == "Patchworks"]
     with open("date_cur.txt", "w") as f:
         if str(runs[0]['id']) != str(run_id):
             f.write(runs[0]['created_at'])
         else:
             f.write(runs[1]['created_at'])
 
+def get_run_id(runs, run_id: str)
+    with open("run_id.txt", "w") as f:
+        if str(runs[0]['id']) != str(run_id):
+            f.write(runs[0]['id'])
+        else:
+            f.write(runs[1]['id'])
 
 def main():
     args = parse_arguments()
     runs = get_workflow_runs(args.token, args.repo)
+    if runs == None:
+        print("Server failure")
+        return
```

Comment:
Should we throw an error here so the github ui shows an x?

Reply:
I'll exit 1